src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -25,22 +25,185 @@
 #include <boost/wave/cpplexer/cpp_lex_token.hpp>
 #include <boost/wave/cpplexer/cpp_lex_iterator.hpp>
 
-#define TCPP_IMPLEMENTATION
-#include <tcpp/source/tcppLibrary.hpp>
-#undef TCPP_IMPLEMENTATION
+//#define TCPP_IMPLEMENTATION
+//#include <tcpp/source/tcppLibrary.hpp>
+//#undef TCPP_IMPLEMENTATION
 
 using namespace nbl;
 using namespace nbl::asset;
 using Microsoft::WRL::ComPtr;
 
 static constexpr const wchar_t* SHADER_MODEL_PROFILE = L"XX_6_7";
 
+
+
 namespace nbl::asset::hlsl::impl
 {
     struct DXC {
         ComPtr<IDxcUtils> m_dxcUtils;
         ComPtr<IDxcCompiler3> m_dxcCompiler;
     };
+
+
+    // for including builtins 
+    struct load_file_or_builtin_to_string
+    {
+        template <typename IterContextT>
+        class inner
+        {
+        public:
+            template <typename PositionT>
+            static void init_iterators(IterContextT& iter_ctx,
+                PositionT const& act_pos, boost::wave::language_support language)
+            {
+                typedef typename IterContextT::iterator_type iterator_type;
+
+                std::string s(iter_ctx.filename.begin(), iter_ctx.filename.end());
+                auto icf = iter_ctx.ctx.get_hooks().preprocessOptions.includeFinder; // TODO it's null?
```

Comment:
if its null, you shouldn't even get here, because `locate_include_file` should have failed https://github.com/Devsh-Graphics-Programming/Nabla/pull/608/files#r1399307666

assert its not null